jconstraints-core/src/main/java/gov/nasa/jpf/constraints/smtlibUtility/smtsolver/SMTCMDContext.java

```diff
@@ -98,39 +109,35 @@ public void pop(int n) {
 
   @Override
   public Result solve(Valuation result) {
+    String output = "";
     try {
-      StringBuilder output = new StringBuilder();
-      String line;
       if (!hasCalledAdd) {
         System.out.println("You might want to add some expressions first");
       } else {
         ctx.flush();
-        while (!solverOutput.ready() && p.isAlive()) {
-          Thread.sleep(10);
-        }
-        while (solverOutput.ready() && (line = solverOutput.readLine()) != null) {
-          output.append(line);
-          output.append("\n");
-        }
+        output += readProcessOutput(ctx, solverOutput) + "\n";
```

Comment:
Couldn't this just assign to output and omit the initialization to an empty string?